driver/controller.go

```diff
@@ -206,6 +208,90 @@ func (s controllerService) validateContentSource(ctx context.Context, req *csi.C
 	return nil, "", status.Errorf(codes.InvalidArgument, "not a proper volume source %v", volumeSource)
 }
 
+// CreateSnapshot creates a logical volume snapshot.
+func (s controllerService) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequest) (*csi.CreateSnapshotResponse, error) {
+	// Since the kubernetes snapshots are Read-Only, we set accessType as 'ro' to activate thin-snapshots as read-only volumes
+	accessType := "ro"
+	// Set snaptype as 'thin' to activate thin-snapshots.
+	// TODO: When adding support for thick-snapshots, set this option as configurable.
```

Comment:
for `TODO` pls follow the form `TODO (name)`

Reply:
Done